### PR TITLE
Adding new muon momentum scale calibration option and making code more clear

### PR DESCRIPTION
**Pull request by davidwalter2**
- Adding new option "datagroups.wmass" to be used in the setupCombineWMass.py script instead of the confusing "datagroups.dilepton" which was a superset for wlike and dilepton
- Moving common options into utilities/common
- Adding muon momentum scale calibration option "massfitData_lblMC" which takes the J/Psi massfit calibration for data and the ideal Geometry + lbl for MC
- changed the default pdf set to "msht20"
- Update the setupCombineWMass.py and makeDataMCStackPlot.py to use a dash separated list of axes names to unroll these axes.
- E.g. instead of using "--hists unrolled" we would use "--hists pt-eta"
Added "set_parser_default" convenience function in utilities/common to set the default argument of the parser

### CODE REVIEW

File: utilities/common.py [hunk 1]
```diff
@@ -98,8 +98,18 @@ def common_parser():
...
     parser.add_argument("--no-color-logger", dest="noColorLogger", action="store_true", default=False, help="Do not use logging with colors")
     parser.add_argument("-v", "--verbose", type=int, default=3, choices=[0,1,2,3,4],
-                        help="Set verbosity level with logging, the larger the more verbose (currently only for setup_test_logger enabled with --set-custom-logger)");
-    
+                        help="Set verbosity level with logging, the larger the more verbose (currently only for setup_test_logger enabled with --set-custom-logger)")
+    parser.add_argument("-e", "--era", type=str, choices=["2016PreVFP","2016PostVFP"], help="Data set to process", default="2016PostVFP")
+    parser.add_argument("--muonCorr", type=str, default="massfitData_lblidealMC", 
+        choices=["none", "trackfit_only", "trackfit_only_mctruth", "lbl", "massfit", "massfit_lbl", "massfitData_mctruth"], 
+        help="Type of correction to apply to the muons")
+    parser.add_argument("--muScaleMag", type=float, default=1e-4, help="Magnitude of dummy muon scale uncertainty")
```
Comment (kdlong) [inline utilities/common.py:106]: So these aren't actually valid options for the gen and low pileup scripts. Can you make a new function that is W/Z specific? So we have common_parser and wz_common_parser for example, where the latter calls the former which is a subset

File: wremnants/muon_calibration.py [hunk 2]
```diff
@@ -120,17 +120,18 @@ def define_corrected_muons(df, cvh_helper, jpsi_helper, corr_type, dataset, smea
     if not (dataset.is_data or dataset.name in common.vprocs):
         corr_type = "none" 
 
+    corr_type = corr_type.replace("massfitData","massfit") if dataset.is_data else corr_type.replace("mctruth","lbl")        
+
     muon = "Muon"
-    if "lbl" in corr_type:
+    if "lbl" in corr_type.split("_"):
```
Comment (kdlong) [inline wremnants/muon_calibration.py:126]: Why is this needed?

**Comment by kdlong** [comment]
I fixed the gen script and refactored the parsers in #140. Can you update your PR to the same approach after this is merged?

File: utilities/common.py [hunk 3]
```diff
@@ -97,9 +97,23 @@ def common_parser():
...
+        parser.add_argument("--muScaleMag", type=float, default=1e-4, help="Magnitude of dummy muon scale uncertainty")
+        parser.add_argument("--muScaleBins", type=int, default=1, help="Number of bins for muon scale uncertainty")
+        parser.add_argument("--muonCorrMag", default=1.e-4, type=float, help="Magnitude of dummy muon momentum calibration uncertainty")
+        parser.add_argument("--muonCorrEtaBins", default=1, type=int, help="Number of eta bins for dummy muon momentum calibration uncertainty")
+        parser.add_argument("--bias-calibration", action='store_true', help="Adjust central value by calibration bias hist")
+        parser.add_argument("--smearing", action='store_true', help="Smear pT such that resolution matches data") #TODO change to --no-smearing once smearing is final
+
     commonargs,_ = parser.parse_known_args()
 
+    logger = setup_logger(os.path.basename(__file__), commonargs.verbose, commoargs.color_logger)
```
Comment (kdlong) [inline utilities/common.py:115]: Not sure we want to call it like this. Won't file be utilities/common here instead of the base script? Would make the logger output not very useful

File: scripts/histmakers/w_z_gen_dists.py [hunk 4]
```diff
@@ -13,9 +13,7 @@
 parser.add_argument("--singleLeptonHists", action='store_true', help="Also store single lepton kinematics")
 parser.add_argument("--skip-ew-hists", action='store_true', help="Also store histograms for EW reweighting. Use with --filter horace")
 
-f = next((x for x in parser._actions if x.dest == "filterProcs"), None)
-if f:
-    f.default = common.vprocs
+parser = common.set_parser_default(parser, "filterProcs", None)
```
Comment (kdlong) [inline scripts/histmakers/w_z_gen_dists.py:16]: Should be ```parser = common.set_parser_default(parser, "filterProcs", common.vprocs)```

Comment (davidwalter2) [reply scripts/histmakers/w_z_gen_dists.py]: Thanks!

**Comment by kdlong** [comment]
@ jeyserma the low PU samples are updated to the new PDFs, right? So I think the only change needed then is  ```extended_pdf_datasets = common.vprocs+common.vprocs_lowpu``` in theory_tools.py

**Comment by jeyserma** [comment]
> @jeyserma the low PU samples are updated to the new PDFs, right? So I think the only change needed then is `extended_pdf_datasets = common.vprocs+common.vprocs_lowpu` in theory_tools.py

I checked and indeed the alternative PDFs are in the lowPU nano, so we can change as well the lowPU to msht20.

**Comment by kdlong** [comment]
@davidwalter2 I think changing this line will fix the failures: https://github.com/WMass/WRemnants/blob/dd5998d260ac80a7af7e6e018694b65098b35076/wremnants/theory_tools.py#L112

should be ```extended_pdf_datasets = common.vprocs+common.vprocs_lowpu```

**Comment by davidwalter2** [comment]
> @davidwalter2 I think changing this line will fix the failures:
> 
> https://github.com/WMass/WRemnants/blob/dd5998d260ac80a7af7e6e018694b65098b35076/wremnants/theory_tools.py#L112
> 
> should be `extended_pdf_datasets = common.vprocs+common.vprocs_lowpu`

Okay thanks, I made another commt, let's see

File: wremnants/datasets/datasets2016.py [hunk 5]
```diff
@@ -25,7 +25,8 @@ def makeFilelist(paths, maxFiles=-1, format_args={}):
         filelist.extend(glob.glob(path) if path[:4] != "/eos" else buildXrdFileList(path, "eoscms.cern.ch"))
     return filelist if maxFiles < 0 else filelist[:maxFiles]
 
-def getDatasets(maxFiles=-1, filt=None, mode=None, base_path=None, nanoVersion="v9", prod_tag="TrackFitV718_NanoProdv1"):
+def getDatasets(maxFiles=-1, filt=None, mode=None, base_path=None, nanoVersion="v9", 
+        data_tag="TrackFitV722_NanoProdv1", mc_tag="TrackFitV718_NanoProdv1"):
```
Comment (kdlong) [inline wremnants/datasets/datasets2016.py:29]: Can you revert this to TrackFitV718_NanoProdv1 for data for now? Then we can make a new PR to update it when the new samples are fully complete
